air-quality-backend/src/etl/forecast/forecast_dao.py

```diff
@@ -83,4 +92,26 @@ def fetch_forecast_data(
         ),
     ]
     results = [fetch_cams_data(*params) for params in task_params]
+
+    # convert the mass mixing ratios to mass concentrations
+    # get pressure on model level 137 from surface pressure
+    # https://confluence.ecmwf.int/display/CKB/ERA5%3A+compute+pressure+and+geopotential
+    # +on+model+levels%2C+geopotential+height+and+geometric+height
+    p_half_above = 0 + 0.997630 * results[0]["sp"]
+    p_half_below = 0 + 1.0 * results[0]["sp"]
+    p_ml = (p_half_above + p_half_below) / 2
+    # surface density: rho = p_ml / (R * T)
+    rho = p_ml / (287.0 * results[1]["t"])
+    for result in results:
+        for variable in result.variables:
+            if result[variable].attrs.get("units") == "kg kg**-1":
+                result[variable] *= rho
+                result[variable].attrs["units"] = "kg m**-3"
+                logging.debug(
+                    f"Updated: {variable}, from units: 'kg kg**-1' to 'kg m**-3'."
+                )
+
+    results[0] = results[0].drop_vars(["sp"])
+    results[1] = results[1].drop_vars(["t"])
```

Comment:
Would this live better inside forecast_data.py itself? We have some code in there already that does some stuff around lat/long conversion. 

It would keep this function a little smaller